### PR TITLE
Update release checklist

### PR DESCRIPTION
**Pull request by siladu**
Following on from the 24.10.0 release https://github.com/hyperledger/besu/issues/7710

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -12,25 +12,30 @@ assignees: ''
...
   - [ ] Pass? Go ahead and complete the release process
   - [ ] Fail? Put a message in #besu-release in Discord indicating the release will be aborted because it failed burn-in 
-- [ ] Using the same git sha, create a calver tag for the FULL RELEASE, example format `24.4.0`
-- [ ] Using the FULL RELEASE tag, create a release in github to trigger the workflows. Once published:
+- [ ] Optional: Perform a dry run with https://github.com/consensys/protocols-release-sandbox to test the workflows
+  - [ ] Sync fork
+  - [ ] git checkout <sha of 24.4.0-RC1>
+  - [ ] git tag 24.4.0; git push origin 24.4.0
+  - [ ] Manually run https://github.com/Consensys/protocols-release-sandbox/actions/workflows/draft-release.yml using `main` branch and `24.4.0` tag
+- [ ] Back on besu, using the same git sha as 24.4.0-RC1, create a calver tag for the FULL RELEASE, example format `24.4.0`
```
Comment (macfarla) [inline .github/ISSUE_TEMPLATE/release-checklist.md:29]: maybe step it out here as above
  - [ ] git checkout <sha of 24.4.0-RC1>
  - [ ] git tag 24.4.0; git push origin 24.4.0

Comment (macfarla) [reply .github/ISSUE_TEMPLATE/release-checklist.md:29]: to make it super obvious that yes you are doing the exact same thing again in the main besu repo

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -12,25 +12,30 @@ assignees: ''
...
+- [ ] Manually run https://github.com/hyperledger/besu/actions/workflows/draft-release.yml inputing the FULL RELEASE tag name. Note, this workflow should always be run from `main` branch (this doesn't prevent hotfix tags from being released)
```
Comment (macfarla) [inline .github/ISSUE_TEMPLATE/release-checklist.md:30]:
```suggestion
- [ ] Manually run https://github.com/hyperledger/besu/actions/workflows/draft-release.yml using `main` branch and `24.4.0` tag ie the FULL RELEASE tag name. Note, this workflow should always be run from `main` branch (this doesn't prevent hotfix tags from being released)
```

**Review by macfarla — APPROVED** [review]

**Review by cdivitotawela — APPROVED** [review]
LGTM

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 1 above)
Comment (siladu) [reply .github/ISSUE_TEMPLATE/release-checklist.md:30]: done

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**